### ISSUE TITLE
BdkProgress not implemented fully

### ISSUE THREAD

**Issue by danielnordh** [bitcoindevkit/bdk-swift#5]
**Problem:**
When attempting to use `wallet.sync`, there doesn't appear to be any way to initialize BdkProgress.

In the Rust examples, a `noop_progress()` is used, and in [BDKSwiftSample](https://github.com/futurepaul/BdkSwiftSample), `Progress()` is used, but that gives me a type error: `Argument type 'Progress' does not conform to expected type 'BdkProgress'`

**Expected behavior:**
The progress types from Rust should be available and possible to initialize.

**Comment by danielnordh** [comment]
In a comment on Discord, it was pointed out that the BDKSwiftSample implements a class that extends `BdkProgress` to solve this problem:

```
class Progress : BdkProgress {
    func update(progress: Float, message: String?) {
        print("progress", progress, message as Any)
    }
}
```

While this got me unblocked, I would still be interested to know if this is the expected behavior, or if the swift bindings are missing something.
On a broader note, better getting starting guides for Swift would be helpful.

**Comment by notmandatory** [comment]
Yes this is currently how the `BdkProgress` is meant to work which is the Swift interpretation of how it works in Rust and Kotlin. I agree we need a getting started guide for Swift, it's still very new.  I'm going to tag this issue as a documentation issue. But if you have any suggestions on how you'd like to see the wallet sync progress listener improved we can discuss it here or if you'd like in a separate issue.

**Comment by danielnordh** [comment]
To wrap this issue up, I'd suggest that the class definition above is added to BitcoinDevKit.swift, otherwise everyone needs to do this themselves.
